src/main/java/org/broadinstitute/hellbender/utils/recalibration/covariates/ContextCovariate.java

```diff
@@ -57,36 +56,46 @@ public ContextCovariate(final RecalibrationArgumentCollection RAC){
     }
 
     @Override
-    public void recordValues(final GATKRead read, final SAMFileHeader header, final ReadCovariates values) {
+    public void recordValues(final GATKRead read, final SAMFileHeader header, final ReadCovariates values, final boolean recordIndelValues) {
+
+        final int originalReadLength = read.getLength();
 
         // store the original bases and then write Ns over low quality ones
-        final byte[] originalBases = Arrays.copyOf(read.getBases(), read.getLength());
-        final byte[] strandedBases = getStrandedBytes(read, lowQualTail);
+        final byte[] strandedClippedBases = getStrandedClippedBytes(read, lowQualTail);  //Note: this makes a copy of the read
 
-        final List<Integer> mismatchKeys = contextWith(strandedBases, mismatchesContextSize, mismatchesKeyMask);
-        final List<Integer> indelKeys = contextWith(strandedBases, indelsContextSize, indelsKeyMask);
+        //Note: we're using a non-standard library here because boxing came up on profiling as taking 20% of time in applyBQSR.
+        //IntList avoids boxing
+        final IntList mismatchKeys = contextWith(strandedClippedBases, mismatchesContextSize, mismatchesKeyMask);
```

Comment:
boxing and unboxing is just slow and will always come up in tight-loop situations like this. So I expect this will always improve things. I don't see it as a huge issue - just one dependency and fastutil is a pretty stable library - now adopted in apache projects like parquet.